index.js

```diff
@@ -37,6 +39,38 @@ module.exports.filter = shouldCompress
 
 var cacheControlNoTransformRegExp = /(?:^|,)\s*?no-transform\s*?(?:,|$)/
 
+/**
+ * @const
+ * whether current node version has brotli support
+ */
+var hasBrotliSupport = false
+try {
+  hasBrotliSupport = 'brotli' in require('process').versions
+} catch (ignored) {
+}
+
+var supportedEncodings = hasBrotliSupport
+  ? ['gzip', 'deflate', 'br', 'identity']
+  : ['gzip', 'deflate', 'identity']
+
+var supportedCompressionsNoDeflate = hasBrotliSupport
+  ? ['gzip', 'br']
+  : ['gzip']
+
+var supportedEncodingsNoDeflate = hasBrotliSupport
+  ? ['gzip', 'br', 'identity']
+  : ['gzip', 'identity']
+
+function copyShallow(opts) {
```

Comment:
Pull in a module to do this if it is necessary; let's not copy and paste a snippet that will be flagged as prototype pollution vulnerability.